clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -1030,6 +1030,7 @@ public void assign(Collection<TopicPartition> partitions) {
      * @return The fetched records (may be empty)
      */
     private Map<TopicPartition, List<ConsumerRecord<K, V>>> pollOnce(long timeout) {
+        client.poll(0L);
```

Comment:
With low values of `max.poll.records`, the need for a system call on every `poll()` actually does cause a performance hit. I wonder if we could just expose `maybeTriggerWakeup()` like you had a in a previous iteration?